### PR TITLE
Update vscode-mssql.d.ts to match what's in the vscode-mssql repo

### PR DESCRIPTION
**Pull request by kisantia**
Originally I was just going to make the same changes that were made in https://github.com/microsoft/vscode-mssql/pull/17657, but noticed there were several other differences. These files are supposed to stay in sync, so this updates the vscode-mssql.d.ts file in ADS to match the current state of the file in the vscode-mssql repo (pending a small sql projects `NoneItems` change that needs to be added in https://github.com/microsoft/vscode-mssql/pull/17659).

### CODE REVIEW

File: extensions/types/vscode-mssql.d.ts [hunk 1]
```diff
@@ -730,107 +757,16 @@ declare module 'vscode-mssql' {
 		moveNoneItem(projectUri: string, destinationPath: string, path: string): Promise<ResultStatus>;
 	}
 
-	export interface GetCrossPlatformCompatibilityResult extends ResultStatus {
```
Comment (kisantia) [inline extensions/types/vscode-mssql.d.ts:733]: these were moved farther down the file in the vscode-mssql version of the file, so they're just moved, not deleted

**Comment by Benjin** [comment]
Any reason to copy changes from vscodeMssql -> ADS rather than the other way around?

**Comment by Charles-Gagnon** [comment]
Since the typings file is supposed to be for the originating repo it should usually go this way - it doesn't really make sense to update the typings file if the source repo hasn't actually been updated.

**Review by Charles-Gagnon — APPROVED** [review]

**Comment by Charles-Gagnon** [comment]
Looks like a couple small things to fix in the sql-bindings extension first though

**Review by kburtram — APPROVED** [review]

**Review by cheenamalhotra — APPROVED** [review]

**Pull request merged**